### PR TITLE
Permissions refactor step 2

### PR DESCRIPTION
**Pull request by craigh**
| Q                 | A
| ----------------- | ---
| Bug fix?          | no
| New feature?      | no
| BC breaks?        | no
| Deprecations?     | no
| Fixed tickets     | -
| Refs tickets      | -
| License           | MIT
| Changelog updated | no

## Description
complete refactoring to Core-2.0 standards (e.g. removal of Util classes, etc)

### CODE REVIEW

**Comment by craigh** [comment]
@Guite @cmfcmf please review

File: src/system/PermissionsModule/Resources/public/js/Zikula.Permission.Admin.View.js [hunk 1]
```diff
@@ -40,92 +33,127 @@ var currentDelete, currentInsertBefore;
...
+            }).fail(function(result) {
+                alert(result.status + ': ' + result.statusText);
+            }).always(function() {
+            });
+        }
 
-            $('#editModal').modal();
-        });
+        function updateEditForm(view) {
+            $('#edit-form-container').replaceWith(view).show();
```
Comment (Guite) [inline src/system/PermissionsModule/Resources/public/js/Zikula.Permission.Admin.View.js]: Nice, didn't know that method before :-)

File: src/system/SecurityCenterModule/Helper/PurifierHelper.php [hunk 2]
```diff
@@ -51,14 +51,9 @@ public function getPurifierConfig($args)
...
-
             if (!is_null($savedConfig) && false !== $savedConfig) {
-                $savedConfig = unserialize($savedConfig);
-                foreach ($savedConfig as $section => $values) {
-                    foreach ($values as $k => $v) {
-                        $config->set($section . '.' . $k, $v);
-                    }
-                }
+                /** @var \HTMLPurifier_Config $config */
+                $config = unserialize($savedConfig);
```
Comment (Guite) [inline src/system/SecurityCenterModule/Helper/PurifierHelper.php]: Why this change? This replaces the whole config. The former version did only override single/specific settings.

Comment (craigh) [inline src/system/SecurityCenterModule/Helper/PurifierHelper.php]: because the old method wasn't working - it couldn't iterate over the object (`as $section => $values)`) So it failed. I'm open to better ideas.

File: src/system/PermissionsModule/Api/AdminApi.php [hunk 3]
```diff
@@ -1,550 +0,0 @@
...
-            if (!empty($modinfo['securityschema'])) {
-                $schemas = array_merge($schemas, $modinfo['securityschema']);
-            }
-        }
-        uksort($schemas, 'strnatcasecmp');
-        SecurityUtil::setSchemas($schemas);
-
-        return $schemas;
-    }
-}
```
Comment (cmfcmf) [inline src/system/PermissionsModule/Api/AdminApi.php]: You've just deleted an API file?!

Comment (craigh) [inline src/system/PermissionsModule/Api/AdminApi.php]: I suppose I have, but IMO it isn't useful to other modules so it isn't needed. All the functionality is replaced internally. just because a file (in Core-1.4.x) is _labelled_ as an API doesn't mean it is truly an *external* API. remember these files were seen as mainly collections of the DB interactions not as services to the outside world. Take a look at the provided (now deleted) methods. If you disagree, we can reinstate the file if you feel it is important.
